inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -1529,15 +1533,19 @@ void GNAGraphCompiler::PWLPrimitive(InferenceEngine::CNNLayerPtr layer) {
 
         num_columns = (w_dim_in == 1) ? h_dim_in * c_dim_in : w_dim_in * c_dim_in;
         num_rows = 1;
+
+        if (dnn->new_num_conv_columns) {
+            num_columns = dnn->new_num_conv_columns;
+            dnn->new_num_conv_columns = 0;
+        }
     } else {
         num_columns = FROM_IR_DIM(inputs, 2);
         num_rows = FROM_IR_DIM(inputs, 1);
-    }
 
-    if (dnn->new_num_conv_columns) {
-        num_rows = dnn->new_num_conv_columns;
-        if (inputs->getDims().size() == 4) num_rows /= FROM_IR_DIM(inputs, 3);
-        dnn->new_num_conv_columns = 0;
+        if (dnn->new_num_conv_columns) {
+            num_rows = dnn->new_num_conv_columns;
+            dnn->new_num_conv_columns = 0;
+        }
```

Comment:
well, this part of the code will be modified as some of the tests fail because of the mismatch e.g. this one GNA_NoRegression.smoke_withGNAPlugin_I16_on_rm_cnn4a

Reply:
It is not code duplication - first one assigns value to columns, second one to rows.

Reply:
right, my bad